src/lib/optimizer/strategy/chunk_pruning_rule.cpp

```diff
@@ -239,4 +242,107 @@ std::shared_ptr<TableStatistics> ChunkPruningRule::_prune_table_statistics(const
                                            old_statistics.row_count - static_cast<float>(num_rows_pruned));
 }
 
+std::vector<PredicateChain> ChunkPruningRule::_find_predicate_chains_recursively(
```

Comment:
You mentioned that this cannot be merged with the helper in the BetweenCompositionRule. Could you remind me why that was the case?

Reply:
They are different types of chains